bot/code_review_bot/report/base.py

```diff
@@ -97,27 +80,38 @@ def requires(self, configuration, *keys):
     def calc_stats(self, issues):
         """
         Calc stats about issues:
-        * group issues by class name
+        * group issues by analyzer
         * count their total number
         * count their publishable number
         """
-        groups = itertools.groupby(
-            sorted(issues, key=lambda x: str(x.__class__)), lambda x: x.__class__
-        )
 
-        def stats(items):
+        def by_analyzer(i):
+            return i.analyzer
+
+        groups = itertools.groupby(sorted(issues, key=by_analyzer), by_analyzer)
+
+        def stats(analyzer, items):
             _items = list(items)
+
+            # Lookup the help message, supporting regexes
+            _help = HELP_COMMANDS.get(analyzer)
+            if _help is None:
+                for regex, help_value in HELP_COMMANDS.items():
+                    if isinstance(regex, Pattern) and regex.match(analyzer):
+                        _help = help_value
+                        break
```

Comment:
Nit: instead of breaking, we could check all regexes to make sure we have only one that matches.
```
                    if isinstance(regex, Pattern) and regex.match(analyzer):
                        assert _help is None
                        _help = help_value
```